src/test/java/com/library/controller/BookControllerTest.java

```diff
@@ -0,0 +1,208 @@
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+class BookControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private BookMapper bookMapper;
+
+    @MockBean
+    private BookRepository bookRepository;
+
+    private TestStorage testStorage;
+
+    private static final String MISSING_TYPE = "Not entered data: [authors]";
+
+    private static final String DELETED_SUCCESS = "Book is deleted successfully!";
+
+    @BeforeEach
+    public void init() {
+        testStorage = new TestStorage();
+    }
+
+    @SneakyThrows
+    @Test
+    void getAllBooks() {
+        Pageable paging = PageRequest.of(0, 5);
+        when(bookRepository.findAll(paging)).thenReturn(new PageImpl<Book>(testStorage.getBooks()));
+
+        MockHttpServletResponse response = mockMvc.perform(get("/books")
+                        .param("page", "1")
+                        .param("sizeOnPage", "5"))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andReturn()
+                .getResponse();
+        JSONArray jsonArray = new JSONArray(response.getContentAsString());
+        JSONObject jsonObject = jsonArray.getJSONObject(0);
+
+        assertNotNull(response.getContentAsString());
+        assertEquals(1, jsonObject.get("id"));
+        assertEquals("War and peace", jsonObject.get("title"));
+        assertEquals(3, jsonArray.length());
+        assertEquals("application/json", response.getContentType());
+    }
+
+    @SneakyThrows
+    @Test
+    void getBookById() {
+        when(bookRepository.findById(1L)).thenReturn(Optional.of(testStorage.getBook()));
+
+        MockHttpServletResponse response = mockMvc.perform(get("/books/{id}", "1"))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andReturn()
+                .getResponse();
+        JSONObject jsonObject = new JSONObject(response.getContentAsString());
+
+        assertNotNull(jsonObject);
+        assertEquals("application/json", response.getContentType());
+        assertEquals("War and peace", jsonObject.get("title"));
+        assertEquals(1, jsonObject.get("id"));
+    }
+
+    @SneakyThrows
+    @Test
+    void getBookByIdFailed() {
+        when(bookRepository.findById(1L)).thenThrow(EntityNotFoundException.class);
+
+        MockHttpServletResponse response = mockMvc.perform(get("/books/{id}", "1"))
+                .andDo(print())
+                .andExpect(status().isNotFound())
+                .andReturn()
+                .getResponse();
+        JSONObject jsonObject = new JSONObject(response.getContentAsString());
+
+        assertEquals("application/json", response.getContentType());
+        assertEquals(LocalDateTime.now().format(DateTimeFormatter.ofPattern("dd-MM-yyyy hh:mm:ss")), jsonObject.get("timestamp"));
```

Comment:
1 - Is there some possibility to have this assert failed?
2 - Hardcoded pattern - replace to Constants interface.

Reply:
fixed